x-pack/platform/plugins/shared/sample_data_ingest/server/services/artifact_manager/utils/download.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createWriteStream } from 'fs';
+import { mkdir, open } from 'fs/promises';
+import { pipeline } from 'stream/promises';
+import Path from 'path';
+import fetch from 'node-fetch';
+
+export type MimeType =
+  | 'application/json'
+  | 'application/x-ndjson'
+  | 'text/plain'
+  | 'application/zip'
+  | 'application/gzip'
+  | 'application/x-gzip'
+  | 'application/octet-stream';
+
+const validatePath = (filePath: string): void => {
+  const dirPath = Path.dirname(filePath);
+  const normalizedPath = Path.normalize(filePath);
+
+  if (normalizedPath.startsWith('..') || !normalizedPath.startsWith(dirPath)) {
```

Comment:
I think the first condition in this validation should be `normalizedPath.includes('..')`

Thank you so much for adding these checks - this is excellent work 🚀